### PR TITLE
Parallax but better: Smooth movement cleanup

### PR DESCRIPTION
**Pull request by LemonInTheDark**
## About The Pull Request

Yo. 
I was fucking around with move loops a bit ago, and I realized that parallax looks REALLY bad.
It was kinda fucking me up into thinking space's glide was mistimed, because the background just kept jumpting.

https://user-images.githubusercontent.com/58055496/165656468-6ad829d2-744d-4d38-9494-1684fe233991.mp4

I then remembered a change mso made a few years back, to remove the animation from parallax.(4b04f9012d1763df625e9e4ae75e4cf4bd1f3771)
It was done because it didn't really look that great, and took extra cpu.

So I went and did my best to lower the cost of the existing parallax code, so I could justify the extra work (632c62e54a0e703de482a80a096596d5e5eb14a7)

In the midst of doing this and building the animation code, I discovered that parallax was getting put in a "wait period" state when you crossed some area borders, because of a fucked argument in shuttle code. Fixed that up too, so now the only stutters are from non animates (https://github.com/tgstation/tgstation/commit/b2484a360e5d50fecbbd048eba4c518288a57ecc)

Then, I readded the animations (c79990ce26ba15566287f1c9ec14898b2f9a78ba), taking care to do as little work inside the for loop as possible, and also animating absolute layers, because they do still change, and without it they look stilted.

Now this is nice, but it has a few issues. Mostly that parallax currently sets its glide duration based off the max time between parallax recaulculations. 
This is an issue because what we really want to do is set the glide duration to the length of the move, which we can roughly acquire via glide_size.
This was I assume done to lighten the load on lower performance machines, and save cpu time. This optimization had to go though.

That removal was done in this commit (d0d2905340075bc99ed6a03cc732cc2602a8bb7a), though it includes a few other things that I immediately walked back.

Essentially, gliding has a problem.
If you have two sources of movement acting on you at one time, your glide size will get out of sync, and you'll start to jitter constantly.
This issue is most apparent when walking "with" a conveyor belt, or the tram. You jutter around constantly, and step forward at random intervals.
What I was trying to do was build a system to track glide sizes and their sources, so I could start to use smooth movement for more visible cases, like pressure movement.

The problem is I fucking can't. I could get the delays to sync together nicely, which when it worked created a really nice effect. But it had a problem.
Byond claims that it decides if a movement counts as a glide based off the step_size var, but this doesn't appear to be completely true.
If you move two tiles in one tick, it will treat that as a "jump", and cancel the glide totally. This behavior persisted even when I switched to movement systems that support step_size, and enabled it.

This means we're essentially locked out of good multi source gliding until we can convince lummox to give us a way to set this window. It's really annoying.
Oh and btw if you're from the future and have this tech now, when you add source logic, you'll need to add directions as an arg too, so moving backwards "looks" right and all (Likely by using abs and - + values or something). Also maybe something about diagonals I've not thought about it enough yet.

Anyway that's all just a pipe dream for now, since it's literally impossible to resolve. (ccb05855e6198a052cd9476e481fc41a7f301af4)

Lemme see I made it so parallax only updates once per diagonal move, which saves a lot of cpu time, and also prevents overriding previous transform calls, so things look nicer (ddc6403c1bc4481a68d27b033e815cfd6e577c98)

Oh and I fixed a visual issue that's mostly unrelated to parallax. You know how when you drag an object behind you, and move diagonally, it moves slower then you, and then "jumps"?
It does that because it doesn't have the LONG_DRAG appearance flag. I fixed this by giving everything the LONG_DRAG appearance flag.

Seriously how are there no issue reports about this are our players just desensitized to visual fuckery?

## Why It's Good For The Game

In the process of doing (632c62e54a0e703de482a80a096596d5e5eb14a7) I added `/datum/component/connect_mob_behalf`.
Get fucked

https://user-images.githubusercontent.com/58055496/165658090-de0c9a98-bda6-4b6f-bb58-c4bca2ee0e53.mp4

BY THE WAY:
if you merge this I'd appreciate keeping the commit history, if you wanna be based merge it with the pr desc, they're both useful.

## Changelog

:cl:
qol: Space Parallax will now smoothly animate between steps, based off your movement speed. Please report any jitter.
fix: Parallax will no longer "freeze" when you cross some area boundaries. Looks better now.
fix: Dragging objects diagonally will no longer cause them to visually "jump" to catch up to you. Seriously how did none report this, are we just collectively blind?
/:cl:

### CODE REVIEW

File: code/datums/components/connect_mob_behalf.dm [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+	tracked = null
+
+/datum/component/connect_mob_behalf/proc/handle_tracked_qdel()
+	SIGNAL_HANDLER
+	qdel(src)
+
+/datum/component/connect_mob_behalf/proc/update_signals()
+	unregister_signals()
+	if(!tracked) // Please do not ask, thanks
```
Comment (SuperNovaa41) [inline code/datums/components/connect_mob_behalf.dm:37]: if this is a hard to understand bit of code wouldn't it be better to actually explain it

Comment (LemonInTheDark) [reply code/datums/components/connect_mob_behalf.dm:37]: Being just brutally honest with you I just didn't want to dupe the comment from connect_loc

**Comment by DragonTrance** [comment]
Been having issues where half of the time when I connect while initializations are going on, I have to reconnect to have the parallax load for me
image

image

**Comment by LemonInTheDark** [comment]
round id?

**Comment by DragonTrance** [comment]
Late, but the round ID was 182473 for the one I commented on. It's happened other rounds too

**Comment by LemonInTheDark** [comment]
when you say "connect while initializations are going on" do you mean connect to the server, or observe the round somehow

**Comment by LemonInTheDark** [comment]
Got it. Ought to be fixed now, assuming servers update

**Pull request merged**